### PR TITLE
[Actions] Clearbit

### LINKED ISSUE #2874: [Refactor] Clearbit Actions
**Is there a specific app this action is for?**
`clearbit`

**Please provide a link to the relevant API docs for the specific service / operation.**
https://dashboard.clearbit.com/docs#enrichment-api-company-api-domain-lookup

Clearbit currently returns a 404 when no records are found. Can we catch this case and add a prop that lets the user decide whether they want to handle this as an error or not? For example:
```
errorIfNoRecords: {
  type: "boolean",
  label: "Handle lack of records found as an error?",
  description: "Defaults to `false`. If no records are found, the step will still be considered a success. If `false`, the step will return an error and workflow execution will stop."
  optional: true,
  default: false
}
```

Please also add a `$summary` message and remove the `// legacy hash_id` comment at the top of each action once the refactor is complete.

### PR DESCRIPTION
**Pull request by vellames-turing**
Closes #2874

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/clearbit/clearbit.app.mjs [hunk 1]
```diff
@@ -1,11 +1,157 @@
...
+      description: "The amount of results to return per page.",
+      optional: true,
+    },
+    errorIfNoRecords: {
+      type: "boolean",
+      label: "Handle lack of records found as an error?",
+      description: "Defaults to `false`. If no records are found, the step will still be considered a success. If `false`, the step will return an error and workflow execution will stop.",
+      optional: true,
+      default: false,
+    },
```
Comment (andrewjschuang) [inline components/clearbit/clearbit.app.mjs:69]: Seems a little confusing to me. Can we rephrase this to something like:

```js
ignoreIfNotFound: {
  label: "Ignore Error if Company Name is not found?"
  description: "false - the step will return an error and workflow execution will stop. true -  the step will still be considered a success"
  default: false
}
```

File: components/clearbit/actions/domain-lookup/domain-lookup.mjs [hunk 2]
```diff
@@ -1,53 +1,65 @@
...
+      webhook_url: this.webhookUrl,
+      company_name: this.companyName,
+      linkedin: this.linkedin,
+      twitter: this.twitter,
+      facebook: this.facebook,
     });
+    if (res.error?.message) {
+      $.export("$summary", res.error.message);
+      throw new Error(res.error?.message);
+    }
```
Comment (andrewjschuang) [inline components/clearbit/actions/domain-lookup/domain-lookup.mjs:61]: The API doesn't throw an error if there is one?
Also, I think that we don't `export summary` on errors

File: components/clearbit/actions/find-companies/find-companies.mjs [hunk 3]
```diff
@@ -1,63 +1,46 @@
...
+      query: this.query,
+      maxResults: this.maxResults,
+    };
+    const res = await this.app._paginate(
+      $,
+      this.maxResults || 100,
+      this.app.findCompanies,
+      params,
+    );
+    $.export("$summary", `Found ${res.length} company(ies)`);
```
Comment (andrewjschuang) [inline components/clearbit/actions/find-companies/find-companies.mjs:43]: Can we check the length and export `company/companies` accordingly?

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
@vellames-turing thanks for the submission. I've added some comments.

I think bumping the `minor` version of each component makes more sense since this is not just a bug fix.

Maybe we could create a `_makeRequest()` method and handle the try/catch errors inside it?

```js
_makeRequest() {
  try {
    ...
    if (res.error?.message) {
      throw new Error()
    }
  } catch (e) {
    if (!this.errorIfNoRecords && err.response?.status == 404) {
      $.export("$summary", "No domain found.");
    } else {
      throw e
    }
}
```

Then we don't need to handle this in each component.

**Comment by vellames-turing** [comment]
Hi, there the changes were addressed. I prefer to maintain the `try/catch` in the action because this is very specific of it :)

File: components/clearbit/actions/find-contacts/find-contacts.mjs [hunk 4]
```diff
@@ -0,0 +1,112 @@
+import app from "../../clearbit.app.mjs";
+import options from "../../common/options.mjs";
+
+export default {
+  key: "clearbit-find-contacts",
+  name: "Find Contacts",
+  description: "Find people who work at a specific company. [See the docs here](https://dashboard.clearbit.com/docs#prospector-api)",
+  version: "0.2.0",
```
Comment (feyzullah) [inline components/clearbit/actions/find-contacts/find-contacts.mjs:8]: Is using `version: "0.2.0"` ok here? Since key `clearbit-find-contact` changed to `clearbit-find-contacts`, would it be better to have something like `0.0.1`?

File: components/clearbit/clearbit.app.mjs [hunk 5]
```diff
@@ -1,11 +1,157 @@
...
+      return res;
+    },
+    async paginate(ctx, maxResults, apiRequestFunction, params) {
+      let page = 1;
+      let items = [];
+      while (true) {
+        const res = await apiRequestFunction(ctx, {
+          ...params,
+          page,
+          page_size: 100,
```
Comment (feyzullah) [inline components/clearbit/clearbit.app.mjs:103]: As I see `paginate` is being used in `find-companies` and `find-contacts`. For `find-contacts` [this doc](https://dashboard.clearbit.com/docs#prospector-api-person-search) states that `page_size` is max `20`. Can we set default `page_size` to `20` here?

**Review by feyzullah — COMMENTED** [review]
Hi @vellames-turing I added just a few questions and comments. Can you please check?

**Review by feyzullah — DISMISSED** [review]
LGTM!

**Comment by vellames-turing** [comment]
@js07 Could you help me with these pipeline errors? I don't get it

**Comment by js07** [comment]
@vellames-turing It looks like the PR might be missing changes to the `pnpm-lock.yaml` file. Can you try running these commands (from [this Slack message](https://pipedream-users.slack.com/archives/C01E5KCTR16/p1654103252389289)):

```
git fetch origin
git merge origin/master

# from repo root
npm ci
npx pnpm install -r
git add pnpm-lock.yaml
git commit
git push
```

**Comment by vellames-turing** [comment]
@js07 After execute code

prepare$ husky install
│ husky - Git hooks installed
└─ Done in 368ms
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

components/snowflake
└─┬ snowflake-sdk
  └─┬ asn1.js-rfc2560
    └── ✕ missing peer asn1.js@^5.0.0
Peer dependencies that should be installed:
  asn1.js@^5.0.0

**Comment by vellames-turing** [comment]
Maybe should I do this in another branch? This is happening only in this branch

**Comment by dylburger** [comment]
@vellames-turing Can you make sure you have an `.npmrc` file in the repo root with the following contents:

```
auto-install-peers=true
```

**Comment by vellames-turing** [comment]
> @vellames-turing Can you make sure you have an `.npmrc` file in the repo root with the following contents:
> 
> ```
> auto-install-peers=true
> ```

@dylburger Yeah, it does

**Comment by vellames-turing** [comment]
@dylburger @js07 Looks like even with the warning, all checks have passed now
@michelle0927 could you please approve?

**Review by michelle0927 — APPROVED** [review]
